### PR TITLE
refactor: unify create collection

### LINKED ISSUE #6738: Display the deposit required to create a collection
### What happened?

The Create Collection page has an error message saying a user has insufficient funds to create a collection. It would be nice to show the exact price required (and maybe explain this is not a fee but rather a deposit).

### Please reproduce in steps

https://deploy-preview-6735--koda-beta.netlify.app/ahp/create?tab=Collection

### Expected Behavior

To display the amount required

### What browsers are you seeing the problem on?

Chrome

### At which address did you encounter bug?

_No response_

### Are you logged in?

Yes

### Which wallet you are using?

PolkadotJS

### Screenshots

image

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/ksm/transfer/?target=%3CMy_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address%3E

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #6995: Balance issues on Create collection and Create Token in Basilisk
I wanted to create a collection on Basilisk

1. If I am paying fees in KSM, I want to see my KSM balance, not BSX balance

image

2. I can't mint a collection, even though I have a sufficient balance of both BSX and KSM

image

3. same issue when trying to mint NFT on Basilisk

### LINKED ISSUE #7020: In creating mint on we should calculate costs for existential deposit
For 
- AHP
- AHK
- BSX

Related
---
- #6738

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Related #5939
- [x] Closes #6995
- [x] Closes #6738
- [x] Closes #7020
- [x] introduce new path `/create/collection` and `/prefix/create/collection`

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

## Screenshot 📸

- [ ] My fix has changed UI

### <samp>🤖 Generated by Copilot at 2ecd516</samp>

This pull request adds the functionality to create NFT collections on different blockchains, such as Kusama or Binance Smart Chain. It introduces a new component `CreateCollection.vue` and a new page `_prefix/create/collection.vue` that use various composables, utils, and components to handle the form and the transaction. It also updates the `locales/en.json` file, the `DropUpload` component, the `NeoNftCard` component, and the `useApi` and `useTransaction` composables to support the new feature. Additionally, it adds a new CSS class `monospace` and two new functions `availablePrefixWithIcon` and `depositAmount` in the `utils/chain.ts` file.

### <samp>🤖 Generated by Copilot at 2ecd516</samp>

> _To create a collection of NFTs_
> _You can use the `CreateCollection` piece_
> _It has a form with fields and validations_
> _And supports different blockchains and transactions_
> _With `DropUpload` and `NeoNftCard` for the visuals and fees_

### CODE REVIEW

**Comment by daiagi** [comment]
@preschian 
any chance that this commit:
https://github.com/kodadot/nft-gallery/pull/6974/commits/8ae7b94c9f2f2a528c297e675d91f613e315f1f5

fixes this issue as well?
* #6995

**Comment by preschian** [comment]
> @preschian any chance you would like to tackle this issue?
> 
> * [Balance issues on Create collection and Create Token in Basilisk #6995](https://github.com/kodadot/nft-gallery/issues/6995)
> 
> since you are working on create collection at the moment

sure, would like to check this also

File: components/create/Collection.vue [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+          <p>{{ $t('mint.collection.symbol.message') }}</p>
+          <NeoInput
+            ref="symbolInput"
+            v-model="symbol"
+            :placeholder="$t('mint.collection.symbol.placeholder')"
+            maxlength="10"
+            required
+            expanded />
+        </div>
+      </NeoField>
```
[the remark this answers is not in the document]
Comment (preschian) [reply components/create/Collection.vue]: MD5 need an additional package. how about this one `Math.random().toString(36).slice(2, 7).toUpperCase()`

[the remark this answers is not in the document]
Comment (preschian) [reply components/create/Collection.vue]: Or do we already use that?

**Comment by yangwao** [comment]
Does this fix this one too?
- https://github.com/kodadot/nft-gallery/issues/7020

**Comment by preschian** [comment]
> Does this fix this one too?
> 
> * [In creating mint on we should calculate costs for existential deposit #7020](https://github.com/kodadot/nft-gallery/issues/7020)

Oh TIL, currently only check `balance >= 10 DOT`. What is ED?

anyway, I put constant value in here for the deposit. is there any way to get the deposit amount and that ED from `api.query` stuff? cc @kodadot/internal-dev 

brb exploring https://dotapps.io/ again

https://github.com/kodadot/nft-gallery/blob/6d7084f2e2af8279e62a281368c2c02e0c0b4b04/utils/chain.ts#L60-L78

**Comment by daiagi** [comment]
#6995 still exists when choosing to pay fees with KSM
image


testing with native token (BSX) seems to be working as expected

could this issue be fixed here or push to follow up?
image

**Comment by preschian** [comment]
> #6995 still exists when choosing to pay fees with KSM

oops sorry, I only fix it in the new path `/create/collection`. can you check on this page first? if good I will update it on the old page

File: components/create/Collection.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (vikiival) [reply components/create/Collection.vue]: just call makeSymbol from minimark 👍 

https://github.com/kodadot/packages/blob/9967171b008933590ec4114be8cbf9ba4f1b8c2c/minimark/src/rmrk/shared/identification.ts#L18

**Comment by vikiival** [comment]
> is there any way to get the deposit amount and that ED from api.query stuff? 

Yes

```ts
const collectionDeposit = api.consts.nfts.collectionDeposit.toString()
const itemDeposit = api.consts.nfts.itemDeposit.toString()
```

File: components/create/Collection.vue [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+const createCollection = async () => {
+  let collection:
+    | CollectionToMintBasilisk
+    | CollectionToMintKusama
+    | CollectionToMintStatmine = {
+    file: logo.value,
+    name: name.value,
+    description: description.value,
+    nftCount: unlimited.value ? 0 : max.value,
```
Comment (daiagi) [inline components/create/Collection.vue:235]: `CollectionToMintBasilisk` doesn't have nftCount

could you use `BaseCollectionType`?

File: components/create/Collection.vue [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+      {
+        interaction: Interaction.MINT,
+        urlPrefix: currentChain.value,
+        collection,
+      },
+      currentChain.value
+    )
+  } catch (error) {
+    showNotification(`[ERR] ${error}`, notificationTypes.warn)
+    console.error(error)
```
Comment (daiagi) [inline components/create/Collection.vue:267]: $consola

File: components/create/Collection.vue [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+          prefixAddress,
+          getKusamaAssetId(currentChain.value)
+        )) as PalletBalancesAccountData
+      ).free.toString()
+    }
+  } else {
+    depositSymbol.value = chainInfo?.tokenSymbol?.toHuman()?.[0]
+  }
+
+  balance.value = format(balance.value, chain.tokenDecimals, false)
```
Comment (daiagi) [inline components/create/Collection.vue:308]: composable?

File: pages/_prefix/create/collection.vue [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import CreateCollection from '@/components/create/Collection.vue'
+
+export default {
+  components: {
+    CreateCollection,
+  },
+  layout() {
+    return 'background-layout'
+  },
+}
```
Comment (daiagi) [inline pages/_prefix/create/collection.vue:15]: please add `head()`

see here:

https://github.com/kodadot/nft-gallery/pull/6993

File: pages/create/collection.vue [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import CreateCollection from '@/components/create/Collection.vue'
+
+export default {
+  components: {
+    CreateCollection,
+  },
+  layout() {
+    return 'background-layout'
+  },
+}
```
Comment (daiagi) [inline pages/create/collection.vue:15]: head()

File: utils/chain.ts [hunk 7]
```diff
@@ -38,3 +38,41 @@ export const getAvailablePrefix = (prefix: string): string => {
...
+  return availablePrefixes().map((chain) => {
+    return {
+      ...chain,
+      icon: menus[chain.value] || '',
+    }
+  })
+}
+
+// https://hello.kodadot.xyz/multi-chain/fees/assethub-fees
+export const depositAmount = {
```
Comment (daiagi) [inline utils/chain.ts:61]: please add typing

**Review by daiagi — CHANGES_REQUESTED** [review]
Overall the page works well
a few things here and there

**Comment by prury** [comment]
kusama and rmrk not showing balances/deposit:

image

**Comment by exezbcz** [comment]
image

image
- This issues with the deposit is not changed
-

File: components/create/Collection.vue [hunk 8]
```diff
@@ -0,0 +1,349 @@
+<template>
+  <div class="columns is-centered">
+    <Loader v-model="isLoading" :status="status" />
+    <form class="column is-half" @submit.prevent="createCollection">
+      <h2 class="title is-size-3">
```
Comment (daiagi) [inline components/create/Collection.vue:5]: use h1 please

https://github.com/kodadot/nft-gallery/issues/7002

File: components/create/Collection.vue [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+@import '@/styles/abstracts/variables';
+
+.o-field:not(:last-child) {
+  margin-bottom: 2rem;
+}
+
+.file {
+  margin-bottom: 0;
+}
+
```
Comment (Jarsen136) [inline components/create/Collection.vue:322]:
```suggestion
```

Look like it has not been used. Or we could use `mb-0`

**Comment by preschian** [comment]
Some updates:
- Replace the old page with new components. But I still keep the tab components. I'm unsure how to adjust that because there is no tab in the new design. The new design currently only on `/create/collection` and `/prefix/create/collection`
- added `existentialDeposit` https://polkadot.js.org/docs/substrate/constants#existentialdeposit-u128
- move it to composables

File: components/create/Collection.vue (hunk 9 above)
Comment (preschian) [reply components/create/Collection.vue]: removed

File: components/create/Collection.vue (hunk 8 above)
Comment (preschian) [reply components/create/Collection.vue]: updated

File: utils/chain.ts (hunk 7 above)
Comment (preschian) [reply utils/chain.ts]: removed

File: pages/create/collection.vue (hunk 6 above)
Comment (preschian) [reply pages/create/collection.vue]: added

File: pages/_prefix/create/collection.vue (hunk 5 above)
Comment (preschian) [reply pages/_prefix/create/collection.vue]: added

File: components/create/Collection.vue (hunk 4 above)
Comment (preschian) [reply components/create/Collection.vue]: updated

File: components/create/Collection.vue (hunk 3 above)
Comment (preschian) [reply components/create/Collection.vue]: updated

File: components/create/Collection.vue [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+  } catch (error) {
+    showNotification(`[ERR] ${error}`, notificationTypes.warn)
+    console.error(error)
+  }
+}
+
+watchEffect(async () => {
+  collectionDeposit.value = 0
+  collectionDeposit.value = depositAmount[currentChain.value]?.collection || 0
+})
```
[the remark this answers is not in the document]
Comment (preschian) [reply components/create/Collection.vue]: removed this section

File: components/create/Collection.vue (hunk 2 above)
Comment (preschian) [reply components/create/Collection.vue]: updated

File: components/create/Collection.vue [hunk 11]
```diff
@@ -0,0 +1,349 @@
...
+
+watchEffect(async () => {
+  balance.value = 0
+  depositTokenId.value = 0
+  depositSymbol.value = ''
+
```
[the remark this answers is not in the document]
Comment (preschian) [reply components/create/Collection.vue]: Actually, I intended to use `watchEffect()`, syntax sugar for 

```js
watch([state1, state2, state3], () => {
}, { immediate: true })
```

Using `watch()` reminds me of using `useEffect()` on react, personally, `watchEffect()` looks simpler. Similar to `computed()` stuff

File: components/create/Collection.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (preschian) [reply components/create/Collection.vue]: added

File: components/create/Collection.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (daiagi) [reply components/create/Collection.vue]: i see

File: components/create/CreateCollection.vue [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+              A deposit of
+              <strong>{{ totalCollectionDeposit }} {{ chainSymbol }}</strong>
+              is required to create a collection. Please note, this initial
+              deposit is refundable.
+              <a
+                href="https://hello.kodadot.xyz/multi-chain/fees"
+                target="_blank"
+                class="has-text-link"
+                rel="nofollow noopener noreferrer">
+                Learn more
```
Comment (daiagi) [inline components/create/CreateCollection.vue:137]: some translation work is needed here

File: components/create/CreateCollection.vue [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+    isLoading.value = true
+
+    await transaction(
+      {
+        interaction: Interaction.MINT,
+        urlPrefix: currentChain.value,
+        collection: collection as
+          | CollectionToMintBasilisk
+          | CollectionToMintKusama
+          | CollectionToMintStatmine,
```
Comment (daiagi) [inline components/create/CreateCollection.vue:252]: `BaseCollectionType` ?

**Review by daiagi — COMMENTED** [review]
minor things
Overall code looks (very) good to me

**Review by roiLeo — APPROVED** [review]
let it roll 🪨

**Comment by prury** [comment]
i'm going to give it a works for me since its large and old collection creation will still be working
created a AHK collection and it went well tho

**Review by exezbcz — APPROVED** [review]

**Review by prury — APPROVED** [review]

**Comment by yangwao** [comment]
image

image

**Review by daiagi — APPROVED** [review]

**Comment by yangwao** [comment]
pay 200 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $200 @ 4.29 USD/DOT ~ 46.62 $DOT
🧗 [1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p](https://polkadot.subscan.io/account/1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)
🔗 [0x673fc8f0624f429c966ce781d97d370b72cf3ba5f5b1a7b0964f15c1f3e978ce](https://polkadot.subscan.io/extrinsic/0x673fc8f0624f429c966ce781d97d370b72cf3ba5f5b1a7b0964f15c1f3e978ce)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**